sources/functions/qbittorrent

```diff
@@ -41,48 +36,110 @@ function whiptail_qbittorrent() {
     fi
 }
 
+function qbittorrent_version_info() {
+    case $qbittorrent in
+        4.1.*)
+            libtorrent_branch=RC_1_1
+            libtorrent_minimum_version=1.1.0
+            stdcver="c++14"
+            cryptography_type=built-in
+            ;;
+        4.2.*)
+            libtorrent_branch=RC_1_2
+            libtorrent_minimum_version=1.2.0
+            stdcver="c++14"
+            cryptography_type=built-in
+            ;;
+        4.3.*)
+            libtorrent_branch=RC_1_2
+            libtorrent_minimum_version=1.2.11
+            stdcver="c++17"
+            cryptography_type=built-in
+            ;;
+        4.4.*)
+            libtorrent_branch=RC_2_0
+            libtorrent_minimum_version=2.0
+            stdcver="c++17"
+            cryptography_type=built-in
+            ;;
+        *)
+            echo_error "qBittorrent version should be between 4.1.*.* and 4.3.*.* -- Setup will not proceed."
```

Comment:
4.4 does not currently exist, this code is here only for preparation purposes. It would be foolish to suggest to people that it currently a valid option.

I don't see a way to fix your issue with "4.1.0.0" as versions like 4.1.9.1 exist though they are rare. Throwing regex into the error message will only serve to confuse people. I'm assuming that folks will be looking at a list of releases and choosing the version number they are looking for if they are not using menus.

Reply:
we can probably check for validity of the version before the switchcase with a function that could check the tags available upstream or in the git repository. Then the default case would be more like "Version is valid but not currently handled, please raise a ticket or proceed with the last version's variables"